main/streams/cast.c

```diff
@@ -196,6 +201,10 @@ PHPAPI int _php_stream_cast(php_stream *stream, int castas, void **ret, int show
 	/* synchronize our buffer (if possible) */
 	if (ret && castas != PHP_STREAM_AS_FD_FOR_SELECT) {
 		php_stream_flush(stream);
+		/* Do special handling for user streams as they may not implement the cast method */
+		if (!show_err && is_stream_user_stream(stream)) {
+			castas |= PHP_STREAM_FLAG_SUPPRESS_ERRORS;
+		}
```

Comment:
Why is this done here? This checks `ret` which means it won't be executed during casting check as `ret` is null. This should be only executed during the actual casting so why did you want to suppress it here as well?

Reply:
Doesn't it make more sense to use that fact to check if an error should be emitted or not rather than a flag then?